pkg/operator/ceph/object/config.go

```diff
@@ -229,6 +243,23 @@ func (c *clusterConfig) setFlagsMonConfigStore(rgwConfig *rgwConfig) error {
 			return errors.Wrapf(err, "failed to set %q to %q on %q", flag, val, who)
 		}
 	}
+	if apisToEnable == defaultRgwApis {
+		// revert config to default state in case if some api was disabled and then enabled back in CRD
+		// TODO: should we also restart RGW deployment?
+		err := monStore.Delete(who, "rgw_enable_apis")
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete rgw_enable_apis config on %q", who)
+		}
+	}
+
+	if c.store.Spec.AdminGateway.Enabled() {
+		// enable only admin api for dedicated Admin RGW:
+		whoAdmin := generateCephXUser(rgwConfig.ResourceName + "-admin")
+		err := monStore.Set(whoAdmin, "rgw_enable_apis", "admin")
```

Comment:
if this config is updated for already running RGW instance, then we should restart pod to apply changes. What is the best way to do so?

Reply:
Restarting a pod requires updating the pod spec. A simple change is to set an env var on the pod spec that reflect this setting. If it changes, it will cause the pod to be restarted. 

I still wonder if we can instead set this option with a CLI param. Then it will naturally be part of the pod spec already, and you would never have to worry about the disconnect between this setting in the mon store, and restarting the rgw pod.

Reply:
That's a good thought. 

Also, If we are considering using a model where proposed in huddle today where there is a separate CephObjectStore to be the admin API, this change could be considered unnecessary. We can still enable all APIs on all CephObjectStores and just direct different traffic to different RGWs. 

However, I can also understand how we  might want to limit the APIs enabled to limit the exposed API surface of some daemons. If that's the case, I think we can change the design to instead plan for an ability to enable/disable certain APIs for a CephObjectStore, and we can use Travis's env var suggestion to help restart daemons.